controllers/egressipam/azure/azure.go

```diff
@@ -543,6 +551,15 @@ func getResourceGroupFromResourceID(id string) string {
 	return strings.Split(id, "/")[4]
 }
 
+func shouldAttachInternalLoadBalancerPool(egresIPam *redhatcopv1alpha1.EgressIPAM) bool {
+
+	loadBalancerAttachAnnotation, ok := egresIPam.GetAnnotations()[AzureLoadBalancerAttachAnnotation]
+	if ok {
+		return loadBalancerAttachAnnotation != "false"
```

Comment:
this is now what we had discussed. the possible values were supposed to be: `none`, `same-as-primary-IP`. Any reason?

Reply:
After considering the implementation, felt that it was less explicit and potentially confusing

Reply:
agreed that the boolean less confusing.

Reply:
guys, I asked you to study how Azure networking works and come up with a solution that encompassed all the possibilities. You didn't want to do it and wanted to just fix ARO. So I came up with a annotation scheme that left the door open to further improvements. But you changed that approach to a boolean that essentially says: "is this ARO or is this IPI". I don't want to do it that way.
I'd like to go back to the original design (@sabre1041  now I realize that you changed also the annotation key), which is:
annotation key: `<prefix>/azure-egress-load-balancer`
Allowed values `none`, `same-as-primary-ip`. Default behavior `same-as-primary-ip`.
This allows in the future to support setting the exact name of the egress load balancer should a customer use UPI.

Reply:
@raffaelespazzoli if you don't mind I'd just like to make sure I understand for myself as someone who needs this change in our environment (well the main changes not the specifics of `bool` vs `annotation`). You're wanting to use `annotations` to keep it open for future usage to expand on? I see it right now the 2 `annotations` of `none` and `same-as-primary-ip` are essentially a boolean already, but if a "3rd" option becomes available via a UPI install, this is to allow that functionality? Is that correct?

If so, on the function `shouldAttachInternalLoadBalancerPool` wouldn't that only be needed for ARO specifically, or does this add some functionality for Azure Load Balancers I'm unaware of? I made another comment below on the README asking more or less this question.

Sorry for jumping in with no skin in the game, just curious as someone who uses this operator.

Reply:
you are correct, I want to keep the door open to further improvements. the function `shouldAttachInternalLoadBalancerPool` is needed for any Azure deployment. I don't like the name, it should read something like `getLoadBalancerPools` and encapsulate the logic of interpreting the annotation an returning the right value.